### PR TITLE
[#9255] Migrate remaining features of Instructor Course Enroll page to Angular 

### PR DESCRIPTION
**Pull request by tanhengyeow**
Part of #9255

Summary of PR:
- Migrated `InstructorCourseEnrollAjaxPageAction` to `GetCourseEnrollStudentsAction`
  - Endpoint: `/webapi/course/enroll/students`
  - Requires: 1 non-null query parameters `courseid`.
  - Returns: any error message if the action fails. If the action is successful, data will be loaded into the `Existing students` spreadsheet interface.
  - Added `GetCourseEnrollStudentsActionTest` to test for the new action.
<br>

- New updates to `instructor-course-enroll-page.*`
   - Upon clicking on the `Existing students` spreadsheet, `/courses/enroll/students` is called **once** to fetch existing students data. Data is then loaded into the spreadsheet and the panel would be expanded to show the spreadsheet interface. Subsequent clicks would **not** call any REST endpoint.
2. If there are no existing students, a status message would be displayed and the panel would not be expanded.

Todo:
- [ ] Delete deprecated files after review

### CODE REVIEW

**Comment by wkurniawan07** [comment]
> Subsequent clicks would **not** call any REST endpoint.

Granted that this is a good practice, there should be a way to refresh that panel, particularly after enrolment if the instructor wants to see the updated students list (currently the only way is to refresh, which of course is not that inconvenient).

File: src/test/java/teammates/test/cases/newaction/GetCourseEnrollStudentsActionTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+        loginAsInstructor(instructorId);
+
+        ______TS("Invalid parameters");
+        //no parameters
+        verifyHttpParameterFailure();
+
+        //null course id
+        String[] invalidParams = new String[] {};
+        verifyHttpParameterFailure(invalidParams);
```
Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/newaction/GetCourseEnrollStudentsActionTest.java]: This is the same as `// no parameters`?

File: src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts [hunk 2]
```diff
@@ -191,11 +202,79 @@ export class InstructorCourseEnrollPageComponent implements OnInit {
...
+        this.hotRegisterer.getInstance(this.existingStudentsHOT);
+
+    // Calling REST API only the first time when spreadsheet has no data
+    if (this.getSpreadsheetLength(existingStudentsHOTInstance.getData()) !== 0) {
+      this.loading = false;
+      return;
+    }
+
+    const paramMap: { [key: string]: string } = {
+      courseid: this.courseid,
```
Comment (wkurniawan07) [inline src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts]: Add `user` to enable masquerade mode.

Comment (tanhengyeow) [reply src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts]: I realized `/course/enroll/students` didn't have `user` passed in as a param as well. Added that in as well.

**Pull request merged**